packages/flutter/lib/src/cupertino/colors.dart

```diff
@@ -543,41 +541,105 @@ class CupertinoColors {
 /// effective color (i.e. [CupertinoDynamicColor.value] will come from the effective
 /// color), which is determined by the [BuildContext] it is last resolved against.
 /// If it has never been resolved, the light, normal contrast, base elevation variant
-/// [CupertinoDynamicColor.color] will be the effective color.
-// TODO(LongCatIsLooong): publicize once all Cupertino components have adopted this.
-// {@tool sample}
-//
-// The following snippet will create a [CupertinoButton] whose background color
-// is _lightModeColor_ in light mode but _darkModeColor_ in dark mode.
-//
-//
-// ```dart
-// CupertinoButton(
-//   child: child,
-//   color: CupertinoDynamicColor.withVibrancy(
-//     color: lightModeColor,
-//     darkColor: darkModeColor,
-//   ),
-//   onTap: () => null,
-// )
-// ```
-// {@end-tool}
-//
-// When a Cupertino component is provided with a `CupertinoDynamicColor`, either
-// directly in its constructor, or from an [InheritedWidget] it depends on (for example,
-// [DefaultTextStyle]), the component will automatically resolve the color by calling
-// [CupertinoDynamicColor.resolve], using their own [BuildContext].
-//
-// When used outside of a Cupertino component, such color resolution will not happen
-// automatically. It's essential to call [CupertinoDynamicColor.resolve] with the
-// correct [BuildContext] before using the color to paint, in order to get the
-// desired effect.
+/// [CupertinoDynamicColor.color] will be the default effective color.
+///
+/// Sometimes manually resolving a [CupertinoDynamicColor] is not necessary, because
```

Comment:
This sentence leaves me wondering: How do I as a developer know when to manually resolve and when not?

Reply:
The next few sections are intended to answer that question. Should I replace the `.` at the end with `:` ?